src/config.rs

```diff
@@ -27,12 +27,39 @@ pub struct ServerConfig {
 #[derive(Deserialize, Debug)]
 pub struct RpcMethod {
     pub method: String,
+    pub params: Vec<MethodParam>,
 
     #[serde(default)]
     pub cache: usize,
+}
+
+impl RpcMethod {
+    pub fn inject_block_num(&self) -> Option<usize> {
+        self.params.iter().position(|p| {
+            p == &MethodParam {
```

Comment:
this is a bit magical. maybe just have an extra `inject: true`? and use type to determine if it needs block hash or number